v3.0.0/tasks-schema.json

```diff
@@ -0,0 +1,1449 @@
+                                            "description": "Object defining a sample output type.",
+                                            "type": "object",
+                                            "properties": {
+                                                "output_type_id_params": {
+                                                    "description": "Object containing parameters specifying how samples were drawn.",
+                                                    "examples": [{
+                                                        "output_type_id_params": {
+                                                            "is_required": true,
+                                                            "type": "integer",
+                                                            "min_samples_per_task": 100,
+                                                            "max_samples_per_task": 100
+                                                        }
+                                                    }, {
+                                                        "output_type_id_params": {
+                                                            "is_required": false,
+                                                            "type": "character",
+                                                            "max_length": 6,
+                                                            "min_samples_per_task": 100,
+                                                            "max_samples_per_task": 500,
+                                                            "compound_taskid_set" : ["origin_date", "horizon", "location", "variant"]
+                                                        }
+                                                    }],
+                                                    "type": "object",
+                                                    "properties": {
+                                                        "is_required": {
+                                                            "description": "Boolean. Whether inclusion of samples is required for the submission to be valid",
+                                                            "type": "boolean"
+                                                        },
+                                                        "type": {
+                                                            "description": "Data type of sample indices.",
+                                                            "type": "string",
+                                                            "enum": [
+                                                                "character",
+                                                                "integer"
+                                                            ]
+                                                        },
+                                                        "max_length": {
```

Comment:
I'm not clear on the problem we're trying to solve with `max_length`, especially when hub admins who want to enforce a specific sample index format will need to write a custom validation anyway.

It feels a bit like we're over-optimizing on a guardrail when we don't know if it's truly needed. One more thing to think about validate, etc.

Reply:
Agreed!